shared/src/utils/validators/get-map-validator.ts

```diff
@@ -0,0 +1,49 @@
+export interface ValidationFunctions<T> {
+    keyValidator: (key: string) => boolean;
+    // TODO: I'm not sure I'm happy with this system already.
+    /**
+     * This transformer may return a non-`T` object.
+     */
+    valueTransformer: (value: unknown) => T;
```

Comment:
Remove the `valueTransformer` from the `ValidationFunctions`.
The `valueValidator` and the `consistencyValidator` both get the untransformed value.

A separate helper function can do the transformation and validation in the `valueValidator` if required.

```ts
function isValidObject(type: Constructor, value): boolean {
    return validateSync(plainToInstance(type, value)).length === 0
} 
```

Reply:
I've simplified the whole thing a bit.